cli/util/cancel.go

```diff
@@ -0,0 +1,72 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/nspcc-dev/neo-go/cli/options"
+	"github.com/nspcc-dev/neo-go/cli/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/actor"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/urfave/cli"
+)
+
+func cancelTx(ctx *cli.Context) error {
+	args := ctx.Args()
+	if len(args) == 0 {
+		return cli.NewExitError("transaction hash is missing", 1)
+	} else if len(args) > 1 {
+		return cli.NewExitError("only one transaction hash is accepted", 1)
+	}
+
+	txHash, err := util.Uint256DecodeStringLE(strings.TrimPrefix(args[0], "0x"))
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("invalid tx hash: %s", args[0]), 1)
+	}
+
+	gctx, cancel := options.GetTimeoutContext(ctx)
+	defer cancel()
+
+	c, err := options.GetRPCClient(gctx, ctx)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to create RPC client: %w", err), 1)
+	}
+
+	mainTx, _ := c.GetRawTransactionVerbose(txHash)
+	if !mainTx.Blockhash.Equals(util.Uint256{}) {
+		return cli.NewExitError(fmt.Errorf("transaction %s is already accepted at block %s", txHash, mainTx.Blockhash.StringLE()), 1)
+	}
+	acc, w, err := smartcontract.GetAccFromContext(ctx)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to get account from context to sign the conflicting transaction: %w", err), 1)
+	}
+	defer w.Close()
+
+	if !mainTx.HasSigner(acc.ScriptHash()) {
+		return cli.NewExitError(fmt.Errorf("account %s is not a signer of the conflicting transaction", acc.Address), 1)
+	}
+
+	a, err := actor.NewSimple(c, acc)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to create Actor: %w", err), 1)
+	}
+
+	resHash, _, err := a.SendTunedRun([]byte{byte(opcode.RET)}, []transaction.Attribute{{Type: transaction.ConflictsT, Value: &transaction.Conflicts{Hash: txHash}}}, func(r *result.Invoke, t *transaction.Transaction) error {
+		err := actor.DefaultCheckerModifier(r, t)
+		if err != nil {
+			return err
+		}
+		if t.NetworkFee < mainTx.NetworkFee+1 {
+			t.NetworkFee = mainTx.NetworkFee + 1
+		}
```

Comment:
Good, and now to make the command more customizable we need to add `txctx.GasFlag` to `canceltx` command and then add this provided `gas` flag value to the resulting `t.NetworkFee`. It is needed for cases when `mainTx` is not yet in the pool of the RPC node that you're querying to, in this case the user can directly specify the desired `NetworkFee` value of the conflicting transaction.

Take a look at the example of `--gas` flag usage:
https://github.com/nspcc-dev/neo-go/blob/7a1bf77585a37302ddfae78b2c88ed472f66cbcc/cli/txctx/tx.go#L45-L54

Reply:
And here we also need to check if `mainTx` is not nil.